### ISSUE TITLE
Interesting question, Can we launch an application with Administrator Privileges?

### ISSUE THREAD

**Issue by muruga85** [2gis/Winium.Desktop#114]
I'm trying to automate an application using Winium and there was a need to run the exe as Administrator. Can you please let me know on how to proceed with this?

Here is the sample code I used:

public static DesktopOptions options = new DesktopOptions { ApplicationPath = @".exe path" };
public static WiniumDriverService service = WiniumDriverService.CreateDesktopService(@"D:\Driver\");
public static WiniumDriver driver = new WiniumDriver(service, options);

**Comment by NickAb** [comment]
Have you tried running Winium as administrator? If it does not work, then you can try running Winium as admin and setting `runas` as an app and passing path to the app you want to run as admin as arguments. See https://technet.microsoft.com/en-us/library/bb490994.aspx

I have not tired those options, it is just a guess. Please tell if it works for you.

**Comment by muruga85** [comment]
Thanks a lot for your quick response. I did change the setting manually by accessing Winium.Desktop.Driver --> Right click Properties --> Compatibility --> Run as Administrator. There after any application we launch through Winium code run as administrator. But the setting was made manually. Is there a way to run Winium as Administrator programatically? Please respond.

**Comment by NickAb** [comment]
This depends on how you start Winium.exe. Using `runas` to run Winium should help. See http://stackoverflow.com/questions/8249705/how-to-run-an-application-as-run-as-administrator-from-the-command-prompt and https://technet.microsoft.com/en-us/library/bb490994.aspx

**Comment by muruga85** [comment]
I'm right now starting it as below:

public static DesktopOptions options = new DesktopOptions { ApplicationPath = @".exe path" };
public static WiniumDriverService service = WiniumDriverService.CreateDesktopService(@"D:\Driver\");
public static WiniumDriver driver = new WiniumDriver(service, options);

Can we plug in the runas administrator code here?

**Comment by NickAb** [comment]
You can try setting `ApplicationPath` to `runas` and pass path to your app and all necessary options as an `args` capability value.

**Comment by muruga85** [comment]
I tried but didnt workout as I'm unable to pass runas command as string in the Application path. Below is the one I tried:

public static DesktopOptions options = new DesktopOptions { ApplicationPath = @"runas /user:administrator", Arguments = @"C:\WINDOWS\system32\calc.exe" };

**Comment by GitHubTestUser5** [comment]
@muruga85, please try the following:

public static DesktopOptions options = new DesktopOptions { ApplicationPath = @"C:\Windows\System32\runas.exe", Arguments = @"/user:<computer name>\administrator C:\WINDOWS\system32\calc.exe" };

But here is one more issue, you will need to type the administrator password when prompted. So it won't be an automatic way.

In case if you find a way to launch the tested application with administrator rights, please let me know.

**Comment by SharonJacob** [comment]
Any luck here? I tried to run a batch file as admin. It gives a pop up to confirm to run as admin. I'm not able to get to this pop up

**Comment by muruga85** [comment]
@SharonJacob -  I did change the setting manually by accessing Winium.Desktop.Driver --> Right click Properties --> Compatibility --> Run as Administrator. There after any application we launch through Winium code run as administrator. But the setting was made manually.

This is currently being taken as enhancement and will look forward towards the progress

**Comment by SharonJacob** [comment]
@muruga85  Thank you so much. It's working perfectly fine.

Another question, do you know anything about product developed using Janus? 
My application uses Janus. The UI Spy is not able to locate these elements, do you know why this happens?

**Comment by muruga85** [comment]
No idea on Janus for now. Will try to research around. Can you also try opening up UISPY as administrator and then try to inspect the elements? Let me know if this is fine?

**Comment by SharonJacob** [comment]
@muruga85 The results are still the same. I found something interesting, may be you can help me to build on this.
https://social.msdn.microsoft.com/Forums/windowsdesktop/en-US/c3f142e1-0624-4ec5-a313-482e72d5454d/problem-with-listitem-checkbox-using-automation-ui?forum=windowsaccessibilityandautomation

If we look at the above forum, people say that Inspect is the best tool to identify UI elements while UISpy is best at listening events.
Further Inspect is built on unmanaged UIAutomation(.Net 2.0) and UISpy on managed UIAutomation(.Net 4.0). There are wrappers built to integrate managed and unmanaged UIA. 

I observed that most of the UI elements of my application are visible theough Inspect than UISpy. I am now trying to build my code on unmanaged UIA.   
Not sure if you have considered before. Though it can be of help.

Let me know if you have any thoughts

**Comment by muruga85** [comment]
Not sure of UnManagedUIA and ManagedUIA. But I mostly prefer Inspect than UISpy for my testing activities. Were you able to notice all the tree items using Inspect which you can't do with UISpy?

**Comment by SharonJacob** [comment]
@muruga85  Yup, I was able to see all the elements in Inspect but not in UISpy. Which UIA are you using?

**Comment by muruga85** [comment]
I was able to inspect my application elements both by using UISpy and Inspect tool

**Comment by SharonJacob** [comment]
@muruga85  I tried using unmanaged UIA, It solves the issue.

**Comment by LijuMathew05** [comment]
Hi,
I am trying to automate one windows application using Winium. Everything works fine when running the maven code from eclipse or command line. 
But on trying the same command in Jenkins, the application doesn't come on screen. It just blinks on the taskbar.
Did anyone face this issue?

**Comment by contactkkiran** [comment]
Is there any way yo launch a non exe file using winum?

**Comment by contactkkiran** [comment]
./

**Comment by Maohan1989** [comment]
Hi. I am working on a desktop application using winnium. And i am totaly new to winium, For the desktop application i am unable to inspect elements using any of the inspectors like UISpy, inspect.exe tool etc. What is the issue or am i missing anything. I am able to inspect only login page after that i am not able to do anything. Please can anyone help me on this.

**Comment by Maohan1989** [comment]
The Desktop application is developed using VB Script.

**Comment by contactkkiran** [comment]
No for winium VB doesn’t matter .Is the screen is dot new frame work form ?

**Comment by Maohan1989** [comment]
Hi Kiran. Sorry  I did not get what you are asking. Are you asking about VB.net something other.

**Comment by Maohan1989** [comment]
It is developed purely on VB 6 and java. when click on any of the menus jsp pages will be opened. This is what i got to know from our developer.
